Develop

Haohaoscreamandrun opened the pull request:
陳昊

+ <http://13.236.0.170:8000/api/attractions?page=1>  
+ <http://13.236.0.170:8000/api/attractions?page=0&keyword=劍潭>  
+ <http://13.236.0.170:8000/api/attractions?page=0&keyword=北>  
+ <http://13.236.0.170:8000/api/attraction/10>  
+ <http://13.236.0.170:8000/api/mrts>

Commit:
Feat: attraction api
Changes: Add two new api for attraction
Reason: Complete request in guide before part.1-2

Commit:
Refactor: mydb.cursor(dictionary=true)
Changes: Adjust fetchJSON() function accordingly
Reason: cursor can directly return dict

Commit:
Feat: /api/mrt GET
Changes: Add new function
Reason: According to guide before 1-2

Commit:
Fix: mysql.connector username
Change: Update username according to user created on ubntu
Reason: The username on Ubuntu is jimmy

Commit:
Fix: correct username of mysql
Change: update user in app.py
Reason: user info created on Ubuntu

Commit:
Docs: Update process note

Commit:
Docs: Update process note

Commit:
Docs: Update .ignore

Commit:
Docs: Update process note

Commit:
Docs: Update process note

Commit:
Docs: Update process note

Commit:
Fix: Wrong API
Changes: "/api/attraction/{attractionID}"
Reason: Comply with api guidance

Commit:
Docs: Update Process note

cwpeng reviewed (changes requested):
nextPage field of the last page should be null. For example, nextPage of http://13.236.0.170:8000/api/attractions?page=4 and http://13.236.0.170:8000/api/attractions?page=1&keyword=%E5%8C%97 should be null.

Commit:
Fix: sql = "SELECT * FROM attractions LIMIT 13 OFFSET %s"(and logistic behind)
Changes: retrieve 13 data instead of 12 to tell if there's additional data on next page
Reason: Should return null on nextpage if theres no data.
Resolve PR2 comment 1

Commit:
Docs: Process note update

cwpeng reviewed (approved):
OK. Done.
